docs/tutorials/control/flutter-app.md

```diff
@@ -0,0 +1,736 @@
+      _organization = (await _viam.appClient.listOrganizations()).first;
+      _locations = await _viam.appClient.listLocations(_organization);
+
+      // in Flutter setState tells the UI to rebuild the widgets whos state has changed,
+      // this is how we change from showing a loading screen to a list of values
+      setState(() {
+        _loading = false;
+      });
+    } catch (e) {
+      print(e);
+    }
+  }
+
+  /// This method will navigate to a specific [Location].
+  void _navigateToLocation(Location location) {
+    Navigator.of(context).push(                                                                // <-- uncommented
+        MaterialPageRoute(builder: (_) => LocationScreen(_viam, location)));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(title: const Text('Locations')),
+        // If we are loading, show a loading indicator.
+        // Otherwise, show a list of [Locations]s.
+        body: _loading
+            ? Center(
+                child: const CircularProgressIndicator.adaptive(),
+              )
+            : // Build a list from the [_locations] state.
+            ListView.builder(
+                itemCount: _locations.length,
+                itemBuilder: (_, index) {
+                  final location = _locations[index];
+                  return ListTile(
+                    title: Text(location.name),
+                    onTap: () => _navigateToLocation(location),
+                    trailing: const Icon(Icons.chevron_right),
+                  );
+                }));
+  }
+}
+```
+
+Try running your app.
+Now, when you tap a location, you'll see a list of the smart machines in that location.
+When you tap one of them, you'll see a list of that machine's {{< glossary_tooltip term_id="resource" text="resources" >}}:
+
+{{<gif webm_src="/tutorials/flutter-app/demo.webm" mp4_src="/tutorials/flutter-app/demo.mp4" alt="Rendering of the mobile app. Log in is clicked, then all Clint's locations are shown, Clint's Desk it clicked, all its smart machines are listed, and then desk-robot is clicked, showing a list of components and services belonging to that smart machine." max-width="300px">}}
+
+## Next steps
+
+Congratulations!
+You have successfully made a Flutter app integrated with Viam!
+
+At this point you could customize the robot screen to have more functionality to control the machine or to show data from the robot in neat ways.
+The Viam GitHub repo contains [more example apps](https://github.com/viamrobotics/viam-flutter-sdk/tree/main/example) for your reference.
+
+Flutter is a very powerful tool for building user interfaces.
+You can stylize the look and feel of your app to match your brand.
+Look around [the Flutter documentation](https://docs.flutter.dev/) to learn how.
+
+Before releasing your app to the app stores you’ll need to add an authentication flow to your app instead of adding API keys as environment variables.
```

Comment:
If you want to release it publicly as opposed to having it as a janky personal use app, right?

Reply:
If you released the app publicly with an embedded API key, everyone using that app would be using that robot/location/organization. If you're a company, say selling vacuum robots that connect to viam, you'd want auth so that you could have their app connect to a specific robot within viam rather than all to the same robot or all to the entire organization. So yeah kinda. Let's amend this slightly

```suggestion
If you're planning to release your app for general use, you will need to add an authentication flow to your app instead of adding API keys as environment variables.
```

Reply:
Well yeah--what I mean by janky is you wouldn't include your env variables file in the release but people would have to manually put theirs in which would be a mega pain for most users.